framework/integration/client/nodeconfig/client.go

```diff
@@ -0,0 +1,165 @@
+// +build k8srequired
+
+package nodeconfig
+
+import (
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/e2e-harness/pkg/harness"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/integration/client"
+	"github.com/giantswarm/operatorkit/framework/integration/testresourceset"
+	"github.com/giantswarm/operatorkit/informer"
+)
+
+type Client struct {
+	Framework *framework.Framework
+
+	g8sClient versioned.Interface
+	k8sClient kubernetes.Interface
+}
+
+func New(config client.Config) (*Client, error) {
```

Comment:
Constructors should have their own config structs. Different implementations do different things and have to be configured differently. That being the same here is rather lucky.

Reply:
I actually made the concious decision to do this here, it makes testing _slightly_ more easy as the config is in a different package then `configmap` or `nodeconfig`. Should I change this to be consistent or can this stay like this until we run into issues?

Reply:
Would be cool to get this straight.

Reply:
Done.